reference/conanfile/tools/cmake/cmake_layout.rst

```diff
@@ -19,76 +19,76 @@ For example, this would implement the standard CMake project layout:
         cmake_layout(self)
 
 
-If you want to try it, use the ``conan new hello/0.1 --template=cmake_lib`` template.
+.. note::
 
-It is very important to note that this ``cmake_layout()`` is just calling the ``folders`` and ``cpp``
-attributes described in the (:ref:`layout reference <layout_folders_reference>`).
+    To try it you can use the ``conan new hello/0.1 --template=cmake_lib`` template.
 
-This is the implementation of ``cmake_layout()``:
+The ``cmake_layout()`` sets the ``folders`` and ``cpp``
+attributes described in the (:ref:`layout reference <layout_folders_reference>`).
 
+The assigned values depend on the CMake generator that will be used.
+It can be defined with the ``tools.cmake.cmaketoolchain:generator`` [conf] entry or passing it in the recipe to the
+``cmake_layout(self, cmake_generator)`` function. The assigned values are different if it is a
+multi-config generator (like Visual Studio or Xcode), or a single-config generator (like Unix Makefiles).
 
-.. code:: python
+These are the values assigned by the ``cmake_layout``:
 
-    def cmake_layout(conanfile, generator=None, src_folder="."):
```

Comment:
Ok, it was slightly annoying for me too, because of being too large, and now split in different functions. Maybe we want a simplified example of a generic layout() that is a bit more realistic. Or maybe showing the ``vs_layout()`` or something, to illustrate what type of things can be done.